builtin/bins/dkron-executor-gcppubsub/gcppubsub.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+	dkplugin "github.com/distribworks/dkron/v3/plugin"
+	dktypes "github.com/distribworks/dkron/v3/plugin/types"
+)
+
+// GCPPubSub plugin publish message to topic when Execute method is called.
+type GCPPubSub struct {
+}
+
+// Execute Process method of the plugin
+// "executor": "gcppubsub",
+// "executor_config": {
+//  "project": "project-id",
+//  "topic": "topic-name",
+//  "data": "aGVsbG8gd29ybGQ=" // Optional. base64 encoded data
+//  "attributes": "{\"hello\":\"world\",\"waka\":\"paka\"}" // JSON serialized attributes
+// }
+func (g *GCPPubSub) Execute(args *dktypes.ExecuteRequest, cb dkplugin.StatusHelper) (*dktypes.ExecuteResponse, error) {
+	out, err := g.ExecuteImpl(args)
+	resp := &dktypes.ExecuteResponse{Output: out}
+	if err != nil {
+		resp.Error = err.Error()
+	}
+	return resp, nil
+}
+
+// ExecuteImpl do http request
+func (g *GCPPubSub) ExecuteImpl(args *dktypes.ExecuteRequest) ([]byte, error) {
+	ctx := context.Background()
+	projectID := args.Config["project"]
+	topicName := args.Config["topic"]
+
+	if projectID == "" {
+		return nil, fmt.Errorf("missing project")
+	}
+
+	if topicName == "" {
+		return nil, fmt.Errorf("missing topic")
+	}
+
+	msg, err := ConfigToPubSubMessage(args.Config)
+	if err != nil{
+		return nil, fmt.Errorf("convert config to pubsub message: %w", err)
+	}
+
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return nil, fmt.Errorf("create client for project %q: %w", projectID, err)
+	}
+	defer func() {
+		_ = client.Close()
+	}()
+
+	topic := client.Topic(topicName)
+	res := topic.Publish(ctx, msg)
+	serverID, err := res.Get(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("publish message to topic %q: %w", topicName, err)
+	}
+	return []byte(serverID), nil
+}
+
+func ConfigToPubSubMessage(config map[string]string) (*pubsub.Message, error) {
```

Comment:
consider making this private
```suggestion
func configToPubSubMessage(config map[string]string) (*pubsub.Message, error) {
```